sw/host/tests/ownership/transfer_lib.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(clippy::bool_assert_comparison)]
+use anyhow::{anyhow, Result};
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::chip::boot_log::BootLog;
+use opentitanlib::chip::boot_svc::{Message, UnlockMode};
+use opentitanlib::chip::helper::{OwnershipActivateParams, OwnershipUnlockParams};
+use opentitanlib::chip::rom_error::RomError;
+use opentitanlib::crypto::ecdsa::EcdsaPrivateKey;
+use opentitanlib::crypto::rsa::RsaPublicKey;
+use opentitanlib::ownership::{
+    ApplicationKeyDomain, KeyMaterial, OwnerApplicationKey, OwnerBlock, OwnerConfigItem,
+    OwnershipKeyAlg,
+};
+use opentitanlib::rescue::serial::RescueSerial;
+
+use std::path::Path;
+
+/// Gets the BootLog.
+pub fn get_boot_log(transport: &TransportWrapper, rescue: &RescueSerial) -> Result<BootLog> {
+    rescue.enter(transport, /*reset=*/ true)?;
+    rescue.get_boot_log()
+}
+
+/// Prepares an UnlockOwnership command, sends it to the chip and gets the response.
+pub fn ownership_unlock(
+    transport: &TransportWrapper,
+    rescue: &RescueSerial,
```

Comment:
Do we have both `transport` and `rescue` because `RescueSerial` could go on different physical transports? For example `RescueSerial` over USB?

Reply:
The rescue implementation needs a reference to the TransportWrapper to perform whatever out-of-band signalling is required to activate the rescue mode and to trigger a reset, if needed.

In the case of serial, no out-of-band signal is required, but a reset is required to force the ROM_EXT to run again.

Reply:
Thanks for the explanation.